sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1786,13 +1788,20 @@ static int sof_card_dai_links_create(struct snd_soc_card *card)
 					dev_err(dev, "failed to create dai link %d\n", link_index);
 					return ret;
 				}
+				if (be_id > largest_sdw_be_id)
+					largest_sdw_be_id = be_id;
```

Comment:
Why not introduce a variable under this for loop and update the be_id, something like that:
```diff
diff --git a/sound/soc/intel/boards/sof_sdw.c b/sound/soc/intel/boards/sof_sdw.c
index ef741907b64d..6be54270e7fd 100644
--- a/sound/soc/intel/boards/sof_sdw.c
+++ b/sound/soc/intel/boards/sof_sdw.c
@@ -1777,15 +1777,24 @@ static int sof_card_dai_links_create(struct snd_soc_card *card)
 				return codec_index;
 
 			for (j = 0; j < codec_info_list[codec_index].dai_num ; j++) {
+				int current_be_id;
+
 				ret = create_sdw_dailink(card, &link_index, dai_links,
 							 sdw_be_num, adr_link,
 							 codec_conf, codec_conf_num,
-							 &be_id, &codec_conf_index,
+							 &current_be_id, &codec_conf_index,
 							 &ignore_pch_dmic, append_dai_type, i, j);
 				if (ret < 0) {
 					dev_err(dev, "failed to create dai link %d\n", link_index);
 					return ret;
 				}
+
+				/*
+				 * Update the be_id to match the highest ID used
+				 * for SDW link
+				 */
+				if (be_id < current_be_id)
+					be_id = current_be_id;
 			}
 
 			if (aggregation && endpoint->aggregated)
@@ -1793,6 +1802,8 @@ static int sof_card_dai_links_create(struct snd_soc_card *card)
 		}
 	}
 
+	/* Increment the be_id for non SDW links to avoid overlap */
+	be_id++;
 SSP:
 	/* SSP */
 	if (!ssp_num)
```